lib/osf-components/addon/components/registries/update-dropdown/component.ts

```diff
@@ -0,0 +1,131 @@
+        case RevisionReviewStates.RevisionInProgress:
+            return 'eye';
+        case RevisionReviewStates.RevisionPendingAdminApproval:
+        case RevisionReviewStates.RevisionPendingModeration:
+            return 'clock';
+        default:
+            return '';
+        }
+    }
+
+    @task
+    @waitFor
+    async getRevisionList() {
+
+        if (!registration){
+            const notRegistrationError = this.intl.t('registries.update_dropdown.not_a_registration_error');
+            return this.toast.error(notRegistrationError);
+        }
+        if (this.args.registration.revisions === null || this.args.registration.revisions === undefined) {
+            return {
+                placeholder: this.intl.t('registries.update_dropdown.no_revisions_error'),
+            };
+        }
+        if (this.args.registration.revisions) {
+            try {
+                const revisions = await this.args.registration.queryHasMany('revisions');
+                this.revisions = revisions.sort();
+                return revisions;
+            } catch (e) {
+                const errorMessage = this.intl.t('registries.update_dropdown.revision_error_message');
+                captureException(e, { errorMessage });
+                this.toast.error(getApiErrorMessage(e), errorMessage);
+            }
+        }
+    }
+
+    @task
+    @waitFor
+    async needsMoreUpdates() {
+        if (!this.revisions) {
+            const errorMessage = this.intl.t('registries.update_dropdown.not_a_revision_error');
+            return this.toast.error(errorMessage);
+        }
+        try {
+            this.args.registration.set('revisionState', RevisionReviewStates.RevisionInProgress);
+            const message = this.intl.t('registries.update_dropdown.revision_more_updates_success');
+            this.toast.success(message);
+        } catch (e) {
+            const errorMessage = this.intl.t('registries.update_dropdown.revision_state_not_set_error');
+            captureException(e, { errorMessage });
+            return this.toast.error(getApiErrorMessage(e), errorMessage);
+        }
+        return this.args.registration;
+    }
+
+    @task
+    @waitFor
+    async approveUpdates() {
+        if (!this.revisions) {
+            const errorMessage = this.intl.t('registries.update_dropdown.not_a_revision_error');
+            return this.toast.error(errorMessage);
+        }
+        try {
+            this.args.registration.set('revisionState', RevisionReviewStates.Approved);
+            const message =  this.intl.t('registries.update_dropdown.revision_approved_success');
+            this.toast.success(message);
+        } catch (e) {
+            const errorMessage = this.intl.t('registries.update_dropdown.revision_state_not_set_error');
+            captureException(e, { errorMessage });
+            return this.toast.error(getApiErrorMessage(e), errorMessage);
+        }
+        return this.args.registration;
```

Comment:
Same here. No need to return `this.args.registration`.

Reply:
Got it, it is now removed.